packages/webpack/src/utils/config.ts

```diff
@@ -149,36 +149,43 @@ export function getBaseWebpackPartial(
   if (options.target === 'node') {
     webpackConfig.output.libraryTarget = 'commonjs';
     webpackConfig.node = false;
-  }
 
-  if (options.compiler !== 'swc' && internalOptions.isScriptOptimizeOn) {
-    webpackConfig.optimization = {
-      sideEffects: true,
-      minimizer: [
-        options.target === 'web'
-          ? new TerserPlugin({
-              parallel: true,
-              terserOptions: {
-                ecma: (internalOptions.esm
-                  ? 2016
-                  : 5) as TerserPlugin.TerserECMA,
-                safari10: true,
-                output: {
-                  ascii_only: true,
-                  comments: false,
-                  webkit: true,
-                },
-              },
-            })
-          : new TerserPlugin({
-              terserOptions: {
-                mangle: false,
-                keep_classnames: true,
+    // could be an object { scripts: boolean; styles: boolean }
+    if (options.optimization === true) {
+      webpackConfig.optimization = {
+        minimize: true,
+        minimizer: [
+          new TerserPlugin({
+            terserOptions: {
+              mangle: false,
+              keep_classnames: true,
+            },
+          }),
+        ],
+        concatenateModules: true,
+      };
+    }
+  } else {
```

Comment:
Could `DefinePlugin` from L132 be moved here and added to `extraPlugins` as well? It would resolve #12108 ;)